client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -0,0 +1,369 @@
+
+/**
+ * @category Client
+ * @module
+ */
+
+const { default: customNotification } = require("../../notificationUtils");
+
+const FOURSPACE = '    ';
+
+
+/**
+ * 
+ * @returns "uniqueId" which is just an increment from the counter in the local storage 
+ */
+const getUniqueId = () => {
+    const newId = JSON.parse(sessionStorage.getItem('idCounter')) + 1;
+    sessionStorage.setItem('idCounter', newId);
+    return newId;
+}
+
+/**
+ * #todo
+ * @returns 
+ */
+const getActivityId = () => `Activity_0ay${getUniqueId()}`
+const getEventId = () => `Event_0ay${getUniqueId()}`
+
+
+/**
+ * @description Splits the robot code into an array and deletes all empty lines 
+ * @param {String} robotCode Code from the Code editor
+ * @returns Robot code w/o empty lines as an array
+ */
+const getRobotCodeAsArray = (robotCode) => {
+    const robotCodeAsArray = robotCode.split('\n');
+    for (let i = 0; i < robotCodeAsArray.length; i += 1) {
+        if (robotCodeAsArray[i] === '') {
+            robotCodeAsArray.splice(i, 1);
+            i -= 1;
+        }
+    }
+    return robotCodeAsArray;
+}
+
+/**
+ * @description checks all lines of the settings section for the right syntax and returns all declared applications as an array
+ * @param {Array} robotCodeSettingsSection all lines from the settings section as an array-entry (typeof string)
+ * @returns Array of all declared applications or undefined if an error occures
+ */
+const getApplicationArray = (robotCodeSettingsSection) => {
+    if (typeof robotCodeSettingsSection === 'undefined') return undefined;
+    const robotCode = robotCodeSettingsSection.slice(1)
+    const availableApplications = JSON.parse(sessionStorage.getItem('availableApplications'))
+    let errorWasThrown;
+
+    robotCode.forEach((line) => {
+        const regexForRpaAlias = new RegExp(`Library +RPA[.][a-zA-Z]+`)
+
+        const elementStartsWithLibrary = line.startsWith('Library ');
+        const rpaAliasIsCorrect = regexForRpaAlias.test(line);
+        const applicationIsAvailable = availableApplications.includes(line.split('RPA.')[1])
+
+        if (!elementStartsWithLibrary) {
+            customNotification('Error', `Every line of the "*** Settings ***" Section has to start with "Library"! \nError location: "${line}"`)
+            errorWasThrown = true;
+            return;
+        }
+        if (!rpaAliasIsCorrect) {
+            customNotification('Error', `Application has to start with "RPA." \nError location: "${line}"`)
+            errorWasThrown = true;
+            return;
+        }
+        if (!applicationIsAvailable) {
+            customNotification('Error', `The Application "${String(line.split('RPA.')[1])}" is currently not supported. `)
+            errorWasThrown = true;
+        }
+    })
+
+    const declaredApplications = (errorWasThrown ? undefined : robotCode.map((line) => line.split('RPA.')[1]))
```

Comment:
Do we need this or could we return right away?

Reply:
You want to skip the assignment to "declaredApplications"? Obviously we don't need it, but I thought it was easier to understand what the slightly cryptic code snippet does with the map function. Therefore the assignment to a constant with a speaking name.

You or @WolfgangDaniel  are welcome to decide how we want to handle this.

Reply:
I'd prefer to return right away

Reply:
Then it's fine with me